### PR TITLE
add gpg signing to cicd workflows

### PR DESCRIPTION
**Pull request by robertlincecum**

### CODE REVIEW

**Comment by robertlincecum** [comment]
don't approve yet

File: .github/workflows/build-on-pr.yml [hunk 1]
```diff
@@ -18,3 +18,6 @@ jobs:
       update_version: false
     secrets:
       GH_PAT: ${{ secrets.GH_PAT }}
+      GPG_PRIVATE_KEY: ${{ secrets.GPG_PRIVATE_KEY2 }}
```
Comment (Djadih) [inline .github/workflows/build-on-pr.yml:21]: Why Key2?

Comment (robertlincecum) [reply .github/workflows/build-on-pr.yml:21]: > Why Key2?

there already was one there and I didn't want to delete it just in case

Comment (Djadih) [reply .github/workflows/build-on-pr.yml:21]: Are there also two of secrets.GPG_KEY_ID?

Comment (robertlincecum) [reply .github/workflows/build-on-pr.yml:21]: > Are there also two of secrets.GPG_KEY_ID?

no

**Review by Djadih — APPROVED** [review]

**Review by gameofpointers — APPROVED** [review]

**Pull request merged**